src/typedarray.rs

```diff
@@ -132,13 +132,20 @@ impl<'a, T: TypedArrayElementCreator + TypedArrayElement> TypedArray<'a, T> {
             return Err(());
         }
 
+        TypedArray::<T>::update(length, data, result);
+        
+        Ok(())
+    }
+    
+    ///  Update an existed JS typed array
+    pub unsafe fn update(length: u32,
+                         data: Option<&[T::Element]>,
+                         result: MutableHandleObject) {
```

Comment:
Rather than making a static method like this, I recommend `pub fn update(&mut self, data: &[T::Element])`.